app/controllers/api_controller.rb

```diff
@@ -1,4 +1,10 @@
 class ApiController < ActionController::Base
   include DeviseTokenAuth::Concerns::SetUserByToken
   protect_from_forgery with: :null_session
+
+  before_action :locale_translate
+
+  def locale_translate
+    I18n.locale = I18n.default_locale || (current_user.locale if current_user.present?)
```

Comment:
I think the logic here is inverted. If I, as a user, have a defined locale, I will want to see the site in that locale, and not in the default of the site. If you use an OR `||` and the first condition returns true (or something evaluated as true), the second condition never executes.
Since ruby 2.3 there is available the safe navigation operator `&.` which sends a message to an object only if the object is != nil. For example `current_user&.local` will return 'es' if the user exists, or nil if it doesn't, but it will not raise an exception.

Try to refactor this with those clues!